subprojects/docs/src/docs/userguide/customTasks.adoc

```diff
@@ -250,6 +250,108 @@ Using Gradle's `IncrementalTaskInputs` is not the only way to create tasks that
+
+[[sec:declaring_task_option]]
+==== Declaring an option
+
+Exposing a new command line parameter for a task property is straightforward. You just have to annotate the corresponding setter method of a property with api:org.gradle.api.tasks.options.Option[]. An option requires a mandatory identifier. Additionally, you can provide an optional description. A task can expose as many command line options as properties available in the class.
+
+Let's have a look at an example to illustrate the functionality. The custom task `UrlVerify` verifies whether a given URL can be resolved by making a HTTP call and checking the response code. The URL to be verified is configurable through the property `url`. The setter method for the property is annotated with api:org.gradle.api.tasks.options.Option[].
+
+++++
+<sample id="stringBasedOption" dir="userguide/tasks/commandLineOption/stringOption/buildSrc/src/main/java" title="Declaring a command line option">
+    <sourcefile file="UrlVerify.java" snippet="custom-task-implementation"/>
+</sample>
+++++
+
+All options declared for a task can be <sec:listing_task_options,rendered as console output>> by running the `help` task.
+
+[[sec:using_task_option_command_line]]
+==== Using an option on the command line
+
+Using an option on the command line has to adhere to the following rules:
+
+- The option uses a double-dash as prefix e.g. `--url`. A single dash does not qualify as valid syntax for a task option.
+- The option parameter follows directly after the task declaration e.g. `verifyUrl --url=http://www.google.com/`.
+- Multiple options of a task can be declared in any order on the command line following the task name.
+
+Getting back to the previous example, the build script creates a task instance of type `UrlVerify` and provides a value from the command line through the exposed option.
+
+++++
+<sample id="taskCommandLineOption" dir="userguide/tasks/commandLineOption/stringOption" title="Using a command line option">
+    <sourcefile file="build.gradle" snippet="task"/>
+    <output args="-q verifyUrl --url=http://www.google.com/"/>
+</sample>
+++++
+
+[[sec:supported_task_option_data_types]]
+==== Supported data types for options
+
+Gradle limits the set of data types that can be used for declaring command line options. The use on the command line differ per type.
+
+`boolean`, `Boolean`::
+Describes an option with the value `true` or `false`. Passing the option on the command line does not require assigning a value. For example `--enabled` equates to `true`. The absence of the option uses the default values assign to the property; that is `false` for `boolean` and `null` for the complex data type.
```

Comment:
Silly clarification, maybe: 

That would mean that `--foo=false` would actually be `true`?

Reply:
Any argument will fail with `Command-line option '--foo' does not take an argument.`.